src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/access/AccessControlService.java

```diff
@@ -49,4 +49,24 @@ public AccessControlResponse getRolesGivenUserId(String userId, String authToken
             roleAssignments
         );
     }
+
+    public String getUser(String authToken) {
+        return idamService.getUserId(authToken);
+    }
+
+    public AccessControlResponse getRolesByActorId(String actorId, String authToken) {
+
+        List<RoleAssignment> roleAssignments = roleAssignmentService.getRolesByActorId(actorId, authToken);
+
+        //Safe-guard
+        if (roleAssignments.isEmpty()) {
+            throw new NoRoleAssignmentsFoundException(
+                "User did not have sufficient permissions to perform this action"
+            );
+        }
+
+        roleAssignments.forEach(role -> log.debug("Response from role assignment service '{}'", role.toString()));
```

Comment:
This seems a bit unnecessary for production since it will loop through the list just to find out it should not log it.

You could try to log the entire object for debugging purposes outside of the loop.
log.debug("Response from role assignment service '{}'", roleAssignments);

Reply:
resolved, removed logging

Reply:
i think this one is still unresolved